components/kyma-environment-broker/common/hyperscaler/account_pool.go

```diff
@@ -67,15 +69,20 @@ func (p *secretBindingsAccountPool) IsSecretBindingDirty(hyperscalerType Type, t
 	return false, nil
 }
 
-func (p *secretBindingsAccountPool) MarkSecretBindingAsDirty(hyperscalerType Type, tenantName string) error {
+func (p *secretBindingsAccountPool) MarkSecretBindingAsDirty(hyperscalerType Type, tenantName string, euAccess bool) error {
 	p.mux.Lock()
 	defer p.mux.Unlock()
 
 	labelSelector := fmt.Sprintf("shared!=true, tenantName=%s,hyperscalerType=%s", tenantName, hyperscalerType)
+	labelSelector = p.addEuAccessSelector(labelSelector, euAccess)
 	secretBinding, err := p.getSecretBinding(labelSelector)
-	if err != nil || secretBinding == nil {
```

Comment:
it can happen that the secret is not existing, it is not marked as used etc. So we should not return error in such cases.